### PR TITLE
Make google provider pyarrow dependency explicit

### LINKED ISSUE #42924: missing pyarrow dependency in google provider?
### Apache Airflow Provider(s)

google

### Versions of Apache Airflow Providers

`6.1` but looks it's the same in `>10`

### Apache Airflow version

2.2

### Operating System

tested linux, macos

### Deployment

Virtualenv installation

### Deployment details

linux/macos & `uv pip` to install the packages

### What happened

I dont know if im missing some obvious reason for this, but `pyarrow` is not specified as a dependency for the google provider, while it definetly depends on it: https://github.com/apache/airflow/blob/main/providers/src/airflow/providers/google/cloud/transfers/sql_to_gcs.py#L29

If i do an install in a venv with:
```
dependencies = [
	"apache-airflow==2.2",
	"apache-airflow-providers-google==6.3",
	"google-cloud-bigquery>=1"
]
```

pyarrow won't be installed, and importing from sql_to_gcs will raise a exception

if i remove google-cloud-bigquery, it WILL be installed, i have no idea what causes this behavior since google-cloud-bigquery does list pyarrow as a dependency. But the version is installed is due to `pandas-gbq` and  depending on it

### What you think should happen instead

IMO if a package is used directly, then it's a direct dependency and it shouldn't rely on it being available via indirect dependencies

I can just add pyarrow myself and solve my problem, but i think the dependency should be explicitly defined in the provider

### How to reproduce

```
[project]
name = "tests"
version = "0.1.0"
description = "Add your description here"
readme = "README.md"
requires-python = ">=3.8"
dependencies = [
	"apache-airflow==2.2",
	"apache-airflow-providers-google==6.3",
	"google-cloud-bigquery>=1"
]
```

`uv pip install .`

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by saucoide**
The provider already depends on it directly but it was not listed in provider.yaml, this MR only adds pyarrow to it

Closes: #42924

### CODE REVIEW

File: providers/src/airflow/providers/google/provider.yaml [hunk 1]
```diff
@@ -170,6 +170,7 @@ dependencies:
   # A transient dependency of google-cloud-bigquery-datatransfer, but we
   # further constrain it since older versions are buggy.
   - proto-plus>=1.19.6
+  - pyarrow>=14.0.1
```
Comment (eladkal) [inline providers/src/airflow/providers/google/provider.yaml:174]: Please add a comment in the code explaining why this is needed. This is useful for when we evaluate the dependencies.
```suggestion
  # place explnation
  - pyarrow>=14.0.1
```

Comment (saucoide) [reply providers/src/airflow/providers/google/provider.yaml:174]: @eladkal i'm only making the existing dependency explicit, I added a comment explaining what it is being used for from my understanding (i havent seen this used anywhere else in the provider), but i'm not the author of the code that actually uses it :)

Comment (potiuk) [reply providers/src/airflow/providers/google/provider.yaml:174]: Yeah. The comment look very good.

**Comment by potiuk** [comment]
However you also need to run pre-commit on that change - to get the "provider_dependencies.json" generated for you.

**Comment by saucoide** [comment]
> However you also need to run pre-commit on that change - to get the "provider_dependencies.json" generated for you.

done

**Comment by potiuk** [comment]
the failing check is about kiota new version (same in main) - it will be addressed shortly. Merging.

**Review by potiuk — APPROVED** [review]

**Pull request merged**